lib/matplotlib/backends/_backend_gtk.py

```diff
@@ -109,6 +112,143 @@ def _on_timer(self):
+            self.window.add(self.vbox)
+            self.vbox.show()
+            self.canvas.show()
+            self.vbox.pack_start(self.canvas, True, True, 0)
+        elif gtk_ver == 4:
+            self.window.set_child(self.vbox)
+            self.vbox.prepend(self.canvas)
+
+        # calculate size for window
+        w, h = self.canvas.get_width_height()
+
+        self.toolbar = self._get_toolbar()
+
+        if self.toolmanager:
+            backend_tools.add_tools_to_manager(self.toolmanager)
+            if self.toolbar:
+                backend_tools.add_tools_to_container(self.toolbar)
+
+        if self.toolbar is not None:
+            if gtk_ver == 3:
+                self.toolbar.show()
+                self.vbox.pack_end(self.toolbar, False, False, 0)
+            elif gtk_ver == 4:
+                sw = Gtk.ScrolledWindow(vscrollbar_policy=Gtk.PolicyType.NEVER)
+                sw.set_child(self.toolbar)
+                self.vbox.append(sw)
+            min_size, nat_size = self.toolbar.get_preferred_size()
+            h += nat_size.height
+
+        self.window.set_default_size(w, h)
+
+        self._destroying = False
+        self.window.connect("destroy", lambda *args: Gcf.destroy(self))
+        self.window.connect({3: "delete_event", 4: "close-request"}[gtk_ver],
+                            lambda *args: Gcf.destroy(self))
+        if mpl.is_interactive():
+            self.window.show()
+            self.canvas.draw_idle()
+
+        self.canvas.grab_focus()
+
+    def destroy(self, *args):
+        if self._destroying:
+            # Otherwise, this can be called twice when the user presses 'q',
+            # which calls Gcf.destroy(self), then this destroy(), then triggers
+            # Gcf.destroy(self) once again via
+            # `connect("destroy", lambda *args: Gcf.destroy(self))`.
+            return
+        self._destroying = True
+        self.window.destroy()
+        self.canvas.destroy()
+
+    def show(self):
+        # show the figure window
+        self.window.show()
+        self.canvas.draw()
+        if mpl.rcParams["figure.raise_window"]:
+            meth_name = {
+                3: "get_window", 4: "get_surface"}[Gtk.get_major_version()]
+            if getattr(self.window, meth_name)():
+                self.window.present()
+            else:
+                # If this is called by a callback early during init,
+                # self.window (a GtkWindow) may not have an associated
+                # low-level GdkWindow (on GTK3) or GdkSurface (on GTK4) yet,
+                # and present() would crash.
+                _api.warn_external("Cannot raise window yet to be setup")
+
+    def full_screen_toggle(self):
+        is_fullscreen = {
+            3: lambda w: (w.get_window().get_state()
+                          & Gdk.WindowState.FULLSCREEN),
+            4: lambda w: w.is_fullscreen(),
+        }[Gtk.get_major_version()](self.window)
+        if is_fullscreen:
```

Comment:
Last line is a bit cluttered, so split just a little different:
```suggestion
        is_fullscreen = {
            3: lambda w: (w.get_window().get_state()
                          & Gdk.WindowState.FULLSCREEN),
            4: lambda w: w.is_fullscreen(),
        }[Gtk.get_major_version()]
        if is_fullscreen(self.window):
```

Reply:
sure